go/cmd/vtbackup/cli/vtbackup.go

```diff
@@ -335,6 +340,14 @@ func takeBackup(ctx, backgroundCtx context.Context, topoServer *topo.Server, bac
 	if err != nil {
 		return fmt.Errorf("failed to initialize mysql config: %v", err)
 	}
+	ctx, cancelCtx := context.WithCancel(ctx)
+	backgroundCtx, cancelbackgroundCtx := context.WithCancel(backgroundCtx)
+	mysqld.OnFailure(func(err error) {
+		log.Warning("Cancelling the vtbackup context as MySQL has failed")
+		cancelCtx()
+		cancelbackgroundCtx()
+	})
```

Comment:
We should normally call the context cancel functions no matter what. And given that this does nothing with the error... I wonder if we can't just add an OnTerm function for this. Isn't that all we care about, stopping vtbackup when the mysqld process ends?

Reply:
The log message would then be:
```
log.Warning("Cancelling the vtbackup context as MySQL has terminated")
```

Reply:
Solved via 93d6c5273e0972f3ff56a942a30cadddca29fb6c